File: app/src/main/java/pl/edu/pwr/wojciech/okonski/lab1/lab1/CountBMIforKgM.java
package pl.edu.pwr.wojciech.okonski.lab1.lab1;

public class CountBMIforKgM implements ICountBMI {
    private float MINIMAL_MASS = 10f;
    private float MAXIMAL_MASS = 250f;
    private float MINIMAL_HEIGHT = 0.5f;
    private float MAXIMAL_HEIGHT = 2.5f;

    @Override
    public boolean isMassValid(float mass) {
        return mass > MINIMAL_MASS && mass < MAXIMAL_MASS;
    }

    @Override
    public boolean isHeightValid(float height) {
        return height > MINIMAL_HEIGHT && height < MAXIMAL_HEIGHT;
    }

    @Override
    public float calculateBMI(float mass, float height) {
        if (!isMassValid(mass) || !isHeightValid(height))
            throw new IllegalArgumentException();
        return mass / (height * height);
    }
}

File: app/src/main/java/pl/edu/pwr/wojciech/okonski/lab1/lab1/CountBmiForKgM.kt
package pl.edu.pwr.wojciech.okonski.lab1.lab1

class CountBmiForKgM : ICountBmi {
    private val minimalMass = 10f
    private val maximalMass = 250f
    private val minimalHeight = 0.5f
    private val maximalHeight = 2.5f

    override fun isMassValid(mass: Float): Boolean {
        return mass > minimalMass && mass < maximalMass
    }

    override fun isHeightValid(height: Float): Boolean {
        return height > minimalHeight && height < maximalHeight
    }

    override fun calculateBMI(mass: Float, height: Float): Float {
        if (!isMassValid(mass) || !isHeightValid(height))
            throw IllegalArgumentException()
        return mass / (height * height)
    }
}